sparsesim/src/lib.rs

```diff
@@ -592,7 +579,34 @@ impl QuantumSim {
                             (index, value),
                             |(index, value), (ctls, target, op)| {
                                 if ctls.iter().all(|c| index.bit(*c)) {
-                                    op.as_transform()((index, value), *target)
```

Comment:
Probably better to keep this match in a separate function that takes an OpCode (and performs operation rather than returning a function). perform_transform maybe?